### PR TITLE
XWIKI-20616: Update jstree files and remove reference to throbber.gif

### PR DESCRIPTION
**Pull request by amottier**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-20616

# Changes

## Description

* Update files provided by `jstree` to use the ones provided in version `3.3.16` (version of `jstree` referenced in the `pom.xml`).
* Comment out the reference to `throbber.gif` in `jstree` `mixins.less` file as the gif file is no part of `xwiki-platform-tree-webjar` and rule is override in `tree.less` to use `spinner.gif`.

## Clarifications

* By looking at files such as [`xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less`](https://github.com/xwiki/xwiki-platform/blob/52e78eb4d31a8b449cd9422412edb947df0f71c1/xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less) it appears they were part of version `3.3.5` or earlier (see for example this [diff](https://github.com/vakata/jstree/compare/3.3.5..3.3.6#diff-c6eea180a2a3a75378b36f8ba3d678924453a910d9c7fe1d5a7a0fd6721405ac) of `src/themes/mixins.less` file between version `3.3.5` and `3.3.6`). As [`pom.xml` specifies version `3.3.16`](https://github.com/xwiki/xwiki-platform/blob/52e78eb4d31a8b449cd9422412edb947df0f71c1/xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/pom.xml#L37) this commit update files included in `xwiki-platform-tree-webjar` that are coming from `jstree` project.
* Removing the reference to `throbber.gif` is a way to remove warning in the log that are probably triggered by search bot parsing the CSS. This doesn't seems to trigger any issue in XWiki with normal user interaction.

# Executed Tests

`./mvnw -Psnapshot,dev clean install -f xwiki-platform-core/xwiki-platform-tree/pom.xml` (with Java 17 and Maven 3.9.6)

### CODE REVIEW

**Comment by surli** [comment]
> By looking at files such as [xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less](https://github.com/xwiki/xwiki-platform/blob/52e78eb4d31a8b449cd9422412edb947df0f71c1/xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less) it appears they were part of version 3.3.5 or earlier (see for example this [diff](https://github.com/vakata/jstree/compare/3.3.5..3.3.6#diff-c6eea180a2a3a75378b36f8ba3d678924453a910d9c7fe1d5a7a0fd6721405ac) of src/themes/mixins.less file between version 3.3.5 and 3.3.6). As [pom.xml specifies version 3.3.16](https://github.com/xwiki/xwiki-platform/blob/52e78eb4d31a8b449cd9422412edb947df0f71c1/xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/pom.xml#L37) this commit update files included in xwiki-platform-tree-webjar that are coming from jstree project.

@mflorea apparently you performed the upgrade from jstree 3.3.5 to 3.3.7 in https://github.com/xwiki/xwiki-platform/commit/ced2f2cc1d2141e5170ca76653fca71a45ef801b did you forgot to update those files back then? Maybe we need some specific instructions for upgrading these?

**Comment by amottier** [comment]
@surli @mflorea is there anything I can do on my side to help the merge of this PR? Don't hesitate to reach out if I can assist. Thanks.

**Review by mflorea — CHANGES_REQUESTED** [review]
You're removing customizations that we did on top of jstree. Unfortunately they are not marked so it's not easy to spot them unless you make a diff between our version of the jstree less files and the original jstree less files.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less [hunk 1]
```diff
@@ -101,5 +102,4 @@
 	.jstree-ellipsis { overflow: hidden; }
 	// base height + PADDINGS!
 	.jstree-ellipsis .jstree-anchor { width: calc(100% ~"-" (@base-height + 5px)); text-overflow: ellipsis; overflow: hidden; }
-	.jstree-ellipsis.jstree-no-icons .jstree-anchor { width: calc(100% ~"-" 5px); }
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/mixins.less:104]: Actually it seems this line was removed upstream in version `3.3.6` : https://github.com/vakata/jstree/compare/3.3.5..3.3.6#diff-c6eea180a2a3a75378b36f8ba3d678924453a910d9c7fe1d5a7a0fd6721405ac

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/main.less [hunk 2]
```diff
@@ -4,15 +4,15 @@
 	.jstree-anchor,
 	.jstree-animated,
 	.jstree-wholerow { transition:background-color 0.15s, box-shadow 0.15s; }
-	.jstree-hovered { background:@hovered-bg-color; border-radius:3px; box-shadow:inset 0 0 1px @hovered-shadow-color; }
-	.jstree-context { background:@hovered-bg-color; border-radius:3px; box-shadow:inset 0 0 1px @hovered-shadow-color; }
-	.jstree-clicked { background:@clicked-bg-color; border-radius:3px; box-shadow:inset 0 0 1px @clicked-shadow-color; }
+	.jstree-hovered { background:@hovered-bg-color; border-radius:2px; box-shadow:inset 0 0 1px @hovered-shadow-color; }
+	.jstree-context { background:@hovered-bg-color; border-radius:2px; box-shadow:inset 0 0 1px @hovered-shadow-color; }
+	.jstree-clicked { background:@clicked-bg-color; border-radius:2px; box-shadow:inset 0 0 1px @clicked-shadow-color; }
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/main.less:9]: I agree I'll restore this change.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/main.less [hunk 3]
```diff
@@ -4,15 +4,15 @@
...
 	.jstree-no-icons .jstree-anchor > .jstree-themeicon { display:none; }
 	.jstree-disabled {
-		background:transparent; color:@disabled-color; opacity:@disabled-opacity;
+		background:transparent; color:@disabled-color;
 		&.jstree-hovered { background:transparent; box-shadow:none; }
 		&.jstree-clicked { background:@disabled-bg-color; }
-		> .jstree-icon { filter: url("data:image/svg+xml;utf8,<svg xmlns=\'http://www.w3.org/2000/svg\'><filter id=\'jstree-grayscale\'><feColorMatrix type=\'matrix\' values=\'0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0 0 0 1 0\'/></filter></svg>#jstree-grayscale"); /* Firefox 10+ */ filter: gray; /* IE6-9 */ -webkit-filter: grayscale(100%); /* Chrome 19+ & Safari 6+ */ }
+		> .jstree-icon { opacity:0.8; filter: url("data:image/svg+xml;utf8,<svg xmlns=\'http://www.w3.org/2000/svg\'><filter id=\'jstree-grayscale\'><feColorMatrix type=\'matrix\' values=\'0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0 0 0 1 0\'/></filter></svg>#jstree-grayscale"); /* Firefox 10+ */ filter: gray; /* IE6-9 */ -webkit-filter: grayscale(100%); /* Chrome 19+ & Safari 6+ */ }
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/main.less:15]: I agree I'll restore this change.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less [hunk 4]
```diff
@@ -1,10 +1,10 @@
 // base jstree
 .jstree-node, .jstree-children, .jstree-container-ul { display:block; margin:0; padding:0; list-style-type:none; list-style-image:none; }
 .jstree-node { white-space:nowrap; }
-.jstree-anchor { display:inline-block; white-space:nowrap; padding:0 4px 0 1px; margin:0; vertical-align:top; }
+.jstree-anchor { display:inline-block; color:black; white-space:nowrap; padding:0 4px 0 1px; margin:0; vertical-align:top; }
 .jstree-anchor:focus { outline:0; }
-.jstree-anchor, .jstree-anchor:link, .jstree-anchor:visited, .jstree-anchor:hover, .jstree-anchor:active { text-decoration:none; }
-.jstree-icon { color:@text-color; display:inline-block; text-decoration:none; margin:0; padding:0; vertical-align:top; text-align:center; }
+.jstree-anchor, .jstree-anchor:link, .jstree-anchor:visited, .jstree-anchor:hover, .jstree-anchor:active { text-decoration:none; color:inherit; }
+.jstree-icon { display:inline-block; text-decoration:none; margin:0; padding:0; vertical-align:top; text-align:center; }
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less:7]: I agree I'll restore this change.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less [hunk 5]
```diff
@@ -16,34 +16,6 @@
...
-  }
-  .jstree-anchor,
-  .jstree-anchor:link,
-  .jstree-anchor:visited,
-  .jstree-anchor:hover,
-  .jstree-anchor:active {
-    color: inherit;
-    text-decoration: none;
-  }
-}
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less:31]: I agree I'll restore this change.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less [hunk 6]
```diff
@@ -16,34 +16,6 @@
...
-  }
-  &.jstree-anchor,
-  &.jstree-anchor:link,
-  &.jstree-anchor:visited,
-  &.jstree-anchor:hover,
-  &.jstree-anchor:active {
-    color: inherit;
-    text-decoration: none;
-  }
-}
```
[the remark this answers is not in the document]
Comment (amottier) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/less/jstree/base.less:45]: I agree I'll restore this change.

**Comment by amottier** [comment]
@mflorea hopefully I restored all customization that I accidentally removed. This pull request is again ready for review. Thanks for your help :+1:

**Review by mflorea — APPROVED** [review]
Looks good to me. Thanks a lot for your contribution and for your patience.

**Comment by amottier** [comment]
Thanks a lot for accepting the pull request and for taking the time to review it :+1:

**Pull request merged**